app/views/spree/admin/shared/_order_tabs.html.haml

```diff
@@ -61,10 +61,11 @@
         %li{ class: adjustments_classes }
           = link_to_with_icon 'icon-cogs', t(:adjustments), spree.admin_order_adjustments_url(@order)
 
-        - if feature?(:invoices)
-          - invoices_classes = "active" if current == 'Invoices'
-          %li{ class: invoices_classes }
-            = link_to_with_icon 'icon-cogs', t(:invoices), spree.admin_order_invoices_url(@order)
+        - if feature?(:invoices, spree_current_user)
+          - if @order.can_show_invoice?
```

Comment:
instead of nesting the if statement, you can combine the conditions into one expression.